### PR TITLE
Changing uses of ubuntu 18.04 to 20.04 in workflow actions

### LINKED ISSUE #3202: ubuntu-18.04 Actions image is being deprecated
The deprecation process for the `ubuntu-18.04` GitHub Actions image has begun, and [scheduled brownouts for this image](https://github.com/actions/runner-images/issues/6002) are affecting our CI builds (as they should). We should update our workflows to a newer image.

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #3202

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Review by keyboardDrummer — APPROVED** [review]

File: .github/workflows/release-brew.yml [hunk 1]
```diff
@@ -32,27 +32,29 @@ jobs:
...
-      run: dafny /compile:0 a.dfy
+      run: dafny verify a.dfy
     - name: Check bad
-      run: dafny /compile:0 b.dfy || echo EXPECTED FAILURE
+      run: dafny verify b.dfy || echo EXPECTED FAILURE
     ## Check that a simple program compiles and runs on each supported platform
     - name: Check C# compile
       run: |
-        dafny /compileVerbose:0 /compile:3 /compileTarget:cs /spillTargetCode:3 a.dfy
+        dafny run -target:cs a.dfy > actual.txt
```
Comment (keyboardDrummer) [inline .github/workflows/release-brew.yml:41]: Should be `--target:cs`

Comment (davidcok) [reply .github/workflows/release-brew.yml]: Fixed.

File: .github/workflows/release-brew.yml [hunk 2]
```diff
@@ -31,27 +31,24 @@ jobs:
...
     - name: Check C# compile
       run: |
-        dafny /compileVerbose:0 /compile:3 /compileTarget:cs /spillTargetCode:3 a.dfy
+        dafny run --target:cs a.dfy | tail -n +3 | diff expect.txt -
     - name: Check Go compile
       run: |
-        dafny /compile:3 /spillTargetCode:3 /compileTarget:go a.dfy
-        env GO111MODULE=auto GOPATH=$PWD/a-go go run a-go/src/a.go > actual.txt
-        diff expect.txt actual.txt
+        dafny run --target:go a.dfy | tail -n +3 | diff expect.txt -
```
Comment (cpitclaudel) [inline .github/workflows/release-brew.yml:43]: I'm not sure that these changes are desirable: do we have any other test that checks manual compilation? (I mean spill + compile, not `run`).

**Review by cpitclaudel — CHANGES_REQUESTED** [review]
The change to Ubuntu 20 is good, but I'm not sure about the changes to `run`.  WDYT?

File: .github/workflows/release-brew.yml (hunk 2 above)
Comment (davidcok) [reply .github/workflows/release-brew.yml]: Fair comment --- I reverted these changes. Whether to change these to new CLI and change to 'dafny run' is an independent question that can be decided separately.

**Review by keyboardDrummer — APPROVED** [review]

**Review by cpitclaudel — APPROVED** [review]

**Pull request merged**